rero_ils/modules/loans/api.py

```diff
@@ -706,70 +706,129 @@ def is_notified(self, notification_type=None, counter=0):
         return number_of_reminders_sent(
             self, notification_type=notification_type) > counter
 
-    def create_notification(self, notification_type=None, counter=0):
+    def get_notification_candidates(self, trigger):
+        """Get notifications candidate to create for the notification.
+
+        This function will check the loan and return all possible notifications
+        to create related to it. In some case, a notification must be generate
+        on an other loan than itself (in case of request).
+
+        :param trigger: the fired action trigger (optional)
+        :return a list of tuple. Each tuple represent a notification candidate.
+                Each tuple is composed by 2 elements : the loan object, and the
+                related notification type.
+        """
+        from ..items.api import Item
+
+        candidates = []
+        item = Item.get_record_by_pid(self.item_pid)
+        has_request = item.number_of_requests() > 0
+
+        # AVAILABILITY NOTIFICATION
+        #   If loan just arrived at the library desk we could create
+        #   an AVAILABILITY notification
+        if self.state == LoanState.ITEM_AT_DESK:
+            candidates.append((self, NotificationType.AVAILABILITY))
+
+        # REQUEST & RECALL NOTIFICATION
+        #   When a request is created on a item, the system create a 'pending'
+        #   loan. If the corresponding item is checked-out, we could create a
```

Comment:
```suggestion
        #   loan. If the corresponding item is checked-out, we can create a
```
If you agree on the `could` → `can`, replace it on the other occurrences, thanks.